timrlink.net.Core/Service/TaskService.cs

```diff
@@ -157,23 +157,38 @@ protected async Task AddOrUpdateTask(IDictionary<string, API.Task> existingTaskI
         {
             logger.LogDebug($"Checking Task(Name={task.name}, ExternalId={task.externalId})");
 
-            if (existingTaskIDs.TryGetValue(task.externalId, out var existingTask) && updateTask)
+            existingTaskIDs.TryGetValue(task.externalId, out var existingTask);
+            
+            if (addedTasks.TryGetValue(task.externalId, out var newAddedTask))
             {
-                if (!equalityComparer.Equals(task, existingTask))
+                if (!equalityComparer.Equals(task, newAddedTask))
                 {
-                    logger.LogInformation($"Updating existing Task(Name={task.name}, ExternalId={task.externalId})");
+                    logger.LogInformation(
+                        $"Updating Task that was created before (Name={task.name}, ExternalId={task.externalId})");
                     await timrSync.UpdateTaskAsync(new API.UpdateTaskRequest(task)).ConfigureAwait(false);
                     existingTaskIDs[task.externalId] = task;
                 }
-            }
-            else if (addedTasks.TryGetValue(task.externalId, out var newAddedTask))
+            }  
+            else if (existingTask != null)
             {
-                if (!equalityComparer.Equals(task, newAddedTask))
+                if (!equalityComparer.Equals(task, existingTask))
                 {
-                    logger.LogInformation($"Updating Task that was created before (Name={task.name}, ExternalId={task.externalId})");
-                    await timrSync.UpdateTaskAsync(new API.UpdateTaskRequest(task)).ConfigureAwait(false);
-                    existingTaskIDs[task.externalId] = task;
-                }    
+                    if (updateTask) {
+                        logger.LogInformation(
+                            $"Updating existing Task(Name={task.name}, ExternalId={task.externalId})");
+                        await timrSync.UpdateTaskAsync(new API.UpdateTaskRequest(task)).ConfigureAwait(false);
+                        existingTaskIDs[task.externalId] = task;
+                    }
+                    else
```

Comment:
probably the core change just resolves around adding an else block here right? what are the remaining changes for?

Reply:
Sure. So you like the old style more we had? I thought it's a bit more readable this way. But anyway.

Reply:
I don't understand. Which particular part do you like better? The else here is required for sure.

But why did you use `existingTask != null`? If you already have the boolean of `.TryGetValue()` which is a standard pattern in C#